### ISSUE TITLE
Request for Google Maps API heat maps

### ISSUE THREAD

**Issue by danje** [metabase/metabase#3206]
It would be really useful to be able to use the heatmap feature of Google Maps to visualise data - ideally the ability to represent a "weighting" at a particular lat/long in much the same way as the region maps currently have a "metric" associated with a region:

https://developers.google.com/maps/documentation/javascript/heatmaplayer

Thanks for all your hard work on developing Metabase this far - it's a very useful tool! :+1: :smiley:

**Comment by salsakran** [comment]
@mazameli + @tlrobinson - just out of curiousity did we google remove heat maps for a specific reason?

@danje - we're switching to leaflet maps in a near term release, so it probably won't make sense to re-enable this specific feature. Can you describe the dataset you're using for a heatmap? And specifically, would you still want this if you could upload (or otherwise use) custom GeoJSON maps ?

**Comment by mazameli** [comment]
Hmm. So Tom tells me that his recollection is that heat maps haven't been in the product since before late 2015. There's evidently old code for it lying around that may or may not work?

The intention, both in the docs and in the designs, was to have Heat Map as one of the three options for map type (Region Map, Pin Map, Heat Map), but it clearly got dropped along the way. 😞

**Comment by tlrobinson** [comment]
@salsakran Yeah I've never seen a heatmap option in the UI. I vaguely recall a bit of code for it but AFAIK it's never been wired up since at least I've been working on it.

**Comment by danje** [comment]
@salsakran We have lat/long data on a lot of properties and we would like to be able to plot various information on a per property basis to see where hot spots are at street level - a choropleth map doesn't give us enough detail and I'm not sure that GeoJSON maps would be the right tool for the job. Heat maps would be great! :smile: (I notice that there seems to be some plugins for Leaflet to add a heat map feature). Thanks for your help.

**Comment by edevil** [comment]
I also have a need for heatmaps. I have a dataset that consists of the point of entry (lat/lon) of foreigners in a particular country. There are a lot of data points, but they tend to cluster around areas. Using a pin map is not viable for this number of points.

**Comment by tlrobinson** [comment]
I think the main question for heat maps is whether we should use server-rendered tiles like we currently do for pin maps with > 2000 points, or do it client-side (and if we do that should we have a special endpoint which returns a more efficient representation of points)? As @edevil mentioned, heat maps are most useful when you have a large number of data points.

Either way it probably makes sense to roll this into the port to leaflets

**Comment by x670** [comment]
> I think the main question for heat maps is whether we should use server-rendered tiles like we currently do for pin maps with > 2000 points, or do it client-side (and if we do that should we have a special endpoint which returns a more efficient representation of points)? As @edevil mentioned, heat maps are most useful when you have a large number of data points.

I tried a lot of tools to aggregate geospatial data and it was a big problem before took a look on [kepler.gl](https://github.com/keplergl/kepler.gl/), based on [deck.gl](https://github.com/uber/deck.gl) library for geospatial analysis. It is completely client-side and work faster and smoother than all tools I used before (carto, omni-sky and many other). It is enough fast to process hundreds of thousands points and now look like product-ready solution. There is only few cases, when you need more data, but I didn't found anything better. Also in most cases, when more data need to be processed and visualized, there always is clear and fast solutions to reduce data to reasonable volume without loss of accuracy.

**Comment by hadifarnoud** [comment]
any update on Heatmap? this would be useful for dataset with lots of pins. it makes more sense to visualize with heatmap

**Comment by mazameli** [comment]
@hadifarnoud We've just released version 0.35, and heat maps are currently at the top of [our backlog](https://github.com/metabase/metabase/projects/1) for sometime after version 0.38.

**Comment by FelixLeChat** [comment]
Hey Metabase team. Any update on Heatmap feature?

**Comment by mahendra350** [comment]
is their any possibility to work with google maps api

**Comment by dragonsahead** [comment]
Metabase now supports custom visualizations in v62. Feel free to create the new visualizations and import those in your Metabase instance. You can check these examples:
https://github.com/metabase/custom-viz-thumbs
https://github.com/metabase/custom-viz-calendar-heatmap
